cypress/integration/analytics.test.js

```diff
@@ -15,11 +15,15 @@ beforeEach(() => {
 })
 
 const dataLayerHasEvent = (eventName) => {
-  return cy.window().then((window) => {
-    const allEvents = window.dataLayer.map((evt) => evt.event)
-
-    expect(allEvents).to.include(eventName)
-  })
+  cy.waitUntil(
+    () =>
+      cy.window({ log: false }).then((window) => {
+        const allEvents = window.dataLayer.map((evt) => evt.event)
+
+        return allEvents.includes(eventName)
+      }),
+    { errorMsg: `Event ${eventName} not found` }
+  ).then((assert) => expect(assert).to.be.true)
```

Comment:
```suggestion
  cy.waitUntil(
    () =>
      cy.window({ log: false }).then((window) =>
        window.dataLayer.some((evt) => evt.event === eventName)
      ),
    { errorMsg: `Event ${eventName} not found` }
  ).then((assert) => expect(assert).to.be.true)
```

A dumb question, can you explain the need for this waitUntil?

Reply:
Not dumb at all! 

Some of the Analytics tests were intermittently failing because the assertion to check the data layer was happening too quick, before the event had actually been pushed. I noticed this happening running the tests locally as well, but it usually wasn't a problem because it would succeed out of the 3 tries.

[Here's](https://github.com/vtex-sites/base.store/runs/4899930351#:~:text=%E2%9C%94%EF%B8%8F-,view_item,-event%20add%20view_item) a VTEX Sites Integration Tests example of when it happened.

Since we added it those tests stopped failing. Here's the PR https://github.com/vtex-sites/base.store/pull/231 where it was added, for more context.